tests/phpunit/tests/term/getTerms.php

```diff
@@ -746,6 +746,26 @@ public function test_get_terms_child_of() {
 		$this->assertCount( 1, $terms );
 	}
 
+	/**
+	 * @ticket 55837
+	 */
+	public function test_get_terms_child_of_cache() {
+		$parent = self::factory()->category->create();
+		self::factory()->category->create( array( 'parent' => $parent ) );
+
+		$args  = array(
+			'fields'     => 'ids',
+			'child_of'   => $parent,
+			'hide_empty' => false,
+		);
+		$terms = get_terms( 'category', $args );
+		$this->assertCount( 1, $terms, 'Check count on first response' );
+
+		$terms2 = get_terms( 'category', $args );
+		$this->assertCount( 1, $terms2, 'Check count on cached response' );
+		$this->assertSame( $terms, $terms2, 'Results are the same after caching' );
```

Comment:
```suggestion
		$this->assertCount( 1, $terms2, 'The second call to get_terms() did not return 1 term' );
		$this->assertSame( $terms, $terms2, 'Results are not the same after caching' );
```
The `message` parameter should indicate what went wrong.